### DISCUSSION TITLE
Should an official client have sane defaults, or should millions of users be implicitly told to just build their own client?

### DISCUSSION THREAD

**Discussion by venteto** [bluesky-social/atproto#2002] [Bluesky Client App]
If the latter is the guiding philosophy, is there any point to an official client in the first place?

I can see the mere existence of the availability of custom feed mechanisms becoming a blunt excuse for punting on sane defaults in the official client. Clearly millions of users are not going to go to the trouble (or even have the ability) to implement their own sane defaults. Moreover, third-party tools like SkyFeed have their own bugs (like sometimes not being able to completely delete all traces of a custom feed you no longer want), their own unresponsive developers, and add their own layer of complexity to the overall experience / network / ecosystem / etc.

Is it more prudent to ask the actually motivated and/or able subset of users to go above and beyond to implement sane defaults, or to implement things which _deviate_ from sane defaults?

**Comment by snarfed** [comment]
One key problem is that there often isn't a single objective "sane" default for a given product decision. Reasonable people can differ. #1998 is a good example: some people may not want their own posts in their feeds, but some definitely do. It's not clear that there's a "correct" default there.

Official reference implementations and clients are helpful in many ways, and proliferating settings isn't always a net benefit to users. Some settings are definitely valuable, but not all are worth the development time and effort and the cognivite load on users. Encouraging developers to explore the broader design space is a valid strategy!

**Comment by DavidBuchanan314** [comment]
I'm not sure I understand the question, or rather, I'm only able to parse it as a rhetorical one. I think it's entirely uncontroversial that the official client (or any client, or indeed, any piece of software) should have sensible default behaviours.

What *can* be controversial is deciding what those sensible defaults should be, and reading between the lines it sounds like you disagree with certain design decisions made by the official app. Perhaps it would be more productive to address those design decisions directly?
